File: epoxy-processor/src/main/java/com/airbnb/epoxy/LithoModelAttributeInfo.java
package com.airbnb.epoxy;

import javax.lang.model.element.Element;

class LithoModelAttributeInfo extends AttributeInfo {
  LithoModelAttributeInfo(LithoModelInfo lithoModelInfo,
      Element propElement, HashCodeValidator hashCodeValidator) {
    fieldName = propElement.getSimpleName().toString();
    typeMirror = propElement.asType();
    modelName = lithoModelInfo.getGeneratedName().simpleName();
    modelPackageName = lithoModelInfo.generatedClassName.packageName();
    useInHash = hashCodeValidator.implementsHashCodeAndEquals(typeMirror);
    ignoreRequireHashCode = false;
    generateSetter = true;
    generateGetter = true;
    hasFinalModifier = false;
    packagePrivate = false;
    isGenerated = true;
  }
}

File: epoxy-processor/src/main/java/com/airbnb/epoxy/LithoModelAttributeInfo.kt
package com.airbnb.epoxy

import javax.lang.model.element.Element

internal class LithoModelAttributeInfo(
    lithoModelInfo: LithoModelInfo,
    propElement: Element,
    hashCodeValidator: HashCodeValidator
) : AttributeInfo() {

    init {
        fieldName = propElement.simpleName.toString()
        typeMirror = propElement.asType()
        modelName = lithoModelInfo.generatedName.simpleName()
        packageName = lithoModelInfo.generatedClassName.packageName()
        useInHash = hashCodeValidator.implementsHashCodeAndEquals(typeMirror)
        ignoreRequireHashCode = false
        generateSetter = true
        generateGetter = true
        hasFinalModifier = false
        isPackagePrivate = false
        isGenerated = true
    }
}
